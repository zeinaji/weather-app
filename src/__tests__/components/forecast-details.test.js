import React from "react";
import { cleanup, render } from "@testing-library/react";
import ForecastDetails from "../../components/forecast-details";
import moment from "moment";

afterEach(cleanup);
const forecast = {
  date: 1525046400000,
  temperature: {
    max: 11,
    min: 6,
  },
  wind: {
    speed: 12,
    direction: "mockDirection",
  },
  humidity: 50,
  description: "mockDescription",
  icon: "mockIcon",
};

describe("ForecastDetails", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={forecast} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct props", () => {
    const { getByText } = render(<ForecastDetails forecast={forecast} />);

    const formattedDate = moment(forecast.date).format("ddd Do MMM");
    const date = getByText(formattedDate);
    expect(date.closest("div")).toHaveClass("date-details");

    const maxTemp = getByText(/11/);
    expect(maxTemp.closest("div")).toHaveClass("max-temperature");

    const minTemp = getByText(/6/);
    expect(minTemp.closest("div")).toHaveClass("min-temperature");

    const windSpeed = getByText(/12/);
    expect(windSpeed.closest("div")).toHaveClass("wind-speed");

    const windDirection = getByText("mockDirection");
    expect(windDirection.closest("div")).toHaveClass("wind-direction");
  });
});
