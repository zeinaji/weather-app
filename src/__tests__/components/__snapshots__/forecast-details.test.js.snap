// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
    style="display: flex; width: 100%; justify-content: center;"
  >
    <div
      style="display: flex; flex-direction: column;"
    >
      <div
        class="date-details"
        data-testid="date-id"
      >
        <span
          style="font-size: 50px;"
        >
          Mon 30th Apr
        </span>
      </div>
      <div
        class="max-temperature"
        data-testid="max-temperature-id"
        style="font-size: 25px;"
      >
        Max temperature: 
        <span
          style="opacity: 0.5;"
        >
          11°c
        </span>
      </div>
      <div
        class="min-temperature"
        data-testid="min-temperature-id"
        style="font-size: 25px;"
      >
        Min temperature: 
        <span
          style="opacity: 0.5;"
        >
           6°c
        </span>
      </div>
      <div
        class="humidity"
        data-testid="humidity"
        style="font-size: 25px;"
      >
        Humidity: 
        <span
          style="opacity: 0.5;"
        >
          50
        </span>
      </div>
      <div
        class="wind-speed"
        data-testid="wind-speed"
        style="font-size: 25px;"
      >
        Wind speed: 
        <span
          style="opacity: 0.5;"
        >
          12
        </span>
      </div>
      <div
        class="wind-direction"
        data-testid="wind-direction"
        style="font-size: 25px;"
      >
        Wind direction: 
        <span
          style="opacity: 0.5;"
        >
           mockDirection
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
